javaslang/src/main/java/javaslang/Value.java

```diff
@@ -877,25 +896,29 @@ default CharSeq toCharSeq() {
      * Current items must be comparable
      *
      * @return A new {@link TreeSet}.
-     * @throws ClassCastException if items is not comparable
+     * @throws ClassCastException if items are not comparable
      */
     @SuppressWarnings("unchecked")
     default SortedSet<T> toSortedSet() throws ClassCastException {
-        final Comparator<? super T> comparator;
-        if (this instanceof SortedSet<?>) {
+        final Comparator<T> comparator;
+        if (this instanceof TreeSet<?>) {
```

Comment:
This one is evil - once it was a bugfix to get the comparator and create a new TreeSet instance instead of returning this. But we should return the same instance if the type matches (in this case: TreeSet).

But what if the actual Value type has an underlying comparator, like PriorityQueue and we want to return a SortedSet? Do we need to reuse the Comparator? If so, we might want to have a general interface for collections that have a comparator...